### PR TITLE
Supported Coil as another image loader in addition to Glide and Picasso.

### PR DESCRIPTION
**Pull request by gkashin**
I hereby agree to the terms of the CLA available at: [link](https://yandex.ru/legal/cla/?lang=en)

### Notes
* Added Coil library for loading images and supported:
  * SvgDecoder to load vector images;
  * GifDecoder to load animatable images.
* Changed image loader picker in the settings from boolean switch to slider to be able to choose between 3 options:
Screenshot 2023-11-25 at 15 04 36

### CODE REVIEW

**Comment by lunarstill** [comment]
Hi! Thanks for the pull request. In order to provide contributions you have to adopt our contributor agreement. Check it here, please: https://github.com/divkit/divkit/blob/main/CONTRIBUTING.md

File: client/android/settings.gradle [hunk 1]
```diff
@@ -26,6 +26,7 @@ include ':fonts'
 include ':glide'
 include ':logging'
 include ':picasso'
+include ':coil'
```
Comment (gulevsky) [inline client/android/settings.gradle:29]: Arrange this line in the alphabetical order, please.

File: client/android/div-core/src/main/java/com/yandex/div/core/images/DivImageDownloadCallback.java [hunk 2]
```diff
@@ -16,6 +18,11 @@ public void onSuccess(@NonNull CachedBitmap cachedBitmap) {
         // no implementation
     }
 
+    @UiThread
+    public void onSuccess(@NonNull Drawable drawable) {
```
Comment (gulevsky) [inline client/android/div-core/src/main/java/com/yandex/div/core/images/DivImageDownloadCallback.java:22]: KDoc comment will help to easily distinguish `onSuccess` methods. Can you add one describing the purpose and cases when this method will be called?

**Comment by gkashin** [comment]
Thank you. Addressed.

**Comment by lunarstill** [comment]
Could you squash commits and rebase it to the main branch, please? I'm trying to move your pull request into the internal repository but seems it has some conflicts in `SettingsActionHandler.kt`.

**Comment by gkashin** [comment]
> Could you squash commits and rebase it to the main branch, please? I'm trying to move your pull request into the internal repository but seems it has some conflicts in `SettingsActionHandler.kt`.

Sorry, I guess I need to close this PR and reopen another one as I merged the main branch from git UI and then rebased it locally. So I have a lot of conflicts now which is hard to resolve :)

**Comment by gkashin** [comment]
The new PR: https://github.com/divkit/divkit/pull/37.

Sorry for the churn.

**Pull request closed without merging**